### PR TITLE
Fix process definitions exists check error

### PR DESCRIPTION
**Pull request by zhongjiajie**
This patch fix process definition error when we submit workflow-as-code define

### CODE REVIEW

**Comment by caishunfeng** [comment]
LGTM

**Review by ruanwenjun — APPROVED** [review]
LGTM

File: dolphinscheduler-python/src/main/java/org/apache/dolphinscheduler/server/PythonGatewayServer.java [hunk 1]
```diff
@@ -177,7 +177,7 @@ public Long createOrUpdateProcessDefinition(String userName,
         long projectCode = project.getCode();
         Map<String, Object> verifyProcessDefinitionExists = processDefinitionService.verifyProcessDefinitionName(user, projectCode, name);
 
-        if (verifyProcessDefinitionExists.get(Constants.STATUS) != Status.SUCCESS) {
+        if (verifyProcessDefinitionExists.get(Constants.STATUS) == Status.PROCESS_DEFINITION_NAME_EXIST) {
```
Comment (ruanwenjun) [inline dolphinscheduler-python/src/main/java/org/apache/dolphinscheduler/server/PythonGatewayServer.java]: I see the code, it seems the status may also be `PROJECT_NOT_FOUNT` or `USER_NO_OPERATION_PROJECT_PERM` here?

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: dolphinscheduler-python/src/main/java/org/apache/dolphinscheduler/server/PythonGatewayServer.java (hunk 1 above)
Comment (zhongjiajie) [reply dolphinscheduler-python/src/main/java/org/apache/dolphinscheduler/server/PythonGatewayServer.java]: Hi wenjun, we would create project with assigned user before code running to this line. So I think we nearly zero percent to miss this situation. Unless other process drop project or user object after I create it just now.

Comment (ruanwenjun) [reply dolphinscheduler-python/src/main/java/org/apache/dolphinscheduler/server/PythonGatewayServer.java]: OK, I got you. 
I suggest adding a log if the status is neither SUCCESS and PROCESS_DEFINITION_NAME_EXIST.
This change is not mandatory, It's up to you.

**Review by ruanwenjun — DISMISSED** [review]

File: dolphinscheduler-python/src/main/java/org/apache/dolphinscheduler/server/PythonGatewayServer.java (hunk 1 above)
Comment (zhongjiajie) [reply dolphinscheduler-python/src/main/java/org/apache/dolphinscheduler/server/PythonGatewayServer.java]: Good suggestions, I would add it to codebase, thanks

**Comment by zhongjiajie** [comment]
Hi, @ruanwenjun  I add some log as you said above, PTAL

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Review by lenboo — APPROVED** [review]
+1

**Comment by zhongjiajie** [comment]
Thanks 🎉

**Pull request merged**